src/test/scala/is/hail/stats/StatsSuite.scala

```diff
@@ -62,4 +62,14 @@ class StatsSuite extends SparkSuite {
       for (j <- 0 to 1)
         assert(G(i, j) == G1(i, j))
   }
+
+  @Test def poissonTest() {
+    // compare with R
+    assert(D_==(dpois(5, 10), 0.03783327))
+    assert(qpois(0.3, 10) == 8)
+    assert(qpois(0.3, 10, lowerTail = false) == 12)
+    assert(D_==(ppois(5, 10), 0.06708596))
+    assert(D_==(ppois(5, 10, lowerTail = false), 0.932914))
+    assert(rpois(5) >= 0)
```

Comment:
Add tests that `qpois(ppois(x)) = x` and  `ppois(qpois(ppois(x))) = x`.  Not that it's not true that ppois(qpois(x)) = x in general.


And a loss of precision test, namely find a big enough k like 50 such that:
ppois(10^k, 1) == 1  (precision lost)
and
ppois(10^k, 1, lower_tail=false) < 1 (precision gained)

you can kill the first line and use that k in the second line.